src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -890,15 +1080,11 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     bool are_any_tasks_just_blocked = false;
     for (auto thread = threads.begin(); thread != threads.end(); thread++) {
```

Comment:
This pattern also repeats a lot. We can do better with this:
```suggestion
    for (auto const [thread_id, thread_state]: threads) {
```